fuel-core-interfaces/src/model/block.rs

```diff
@@ -39,18 +54,27 @@ impl FuelBlockHeader {
         hasher.input(self.producer.as_ref());
         hasher.digest()
     }
+
+    pub fn id(&self) -> Bytes32 {
+        if let Some(ref metadata) = self.metadata {
+            metadata.id
+        } else {
+            self.hash()
```

Comment:
shouldn't we replace the metadata once it is calculated?

Reply:
A new block is mostly going to be broadcasted as `Arc` in that case having `&mut` self is not going to work. This else branch is left here as to not break other parts of fuel-core by requiring `recalculate_metadata`  to be called beforehand for the id to be valid.